### ISSUE TITLE
[API Proposal]: synchronous API for creating a Quic listener

### ISSUE THREAD

**Issue by bentoi** [dotnet/runtime#76360]
### Background and motivation

The static `QuicListener.ListenAsync` factory method is asynchronous. However the implementation is synchronous since msquic `ListenerOpen` just allocates ressources. 

Why is it asynchronous? Can I count on this method to always complete synchronously like `Socket.Listen`? 

I'm asking this because I have a transport abstraction which has a synchronous `Listen` API and I'd rather not change it to be asynchronous.

If there's no good reason for it to be asynchronous, I suggest to change it to be synchronous.

### API Proposal

```csharp
public sealed partial class QuicListener : IAsyncDisposable
{
    static QuicListener Listen(QuicListenerOptions options);
}
```

### API Usage

```csharp
var listener = QuicListener.Listen(options);
```

**Comment by rzikm** [comment]
If I remember correctly, the method is asynchronous for consistency with `QuicConnection.ConnectAsync`. CC: @ManickaP 

If you don't want to change your abstraction, you can use `.GetAwaiter().GetResult()` on the returned task. In .NET 7 it will always complete synchronously (however, that may change in future .NET versions).

**Comment by karelz** [comment]
Triage: Looks like a conflict of abstractions between consistency with `QuicConnection.ConnectAsync` and with `Socket.Listen`.
The underlying implementation (msquic, or other in future) might force us to make it async.
We should decide in 8.0 if we want to make changes, or keep it the way it is.

**Comment by ManickaP** [comment]
> If I remember correctly, the method is asynchronous for consistency with `QuicConnection.ConnectAsync`

Yes, the motivation was to keep QUIC APIs consistent. On top of that, `ListenAsync` returns `ValueTask` which has negligible overhead. Especially if we're talking about starting a listener, an operation that is expected to be done only once per program or in a very small number of instances.
You can check `IsCompleted` and get the `Result` synchronously, which is the behavior for .NET 7.0. Although it might change in future releases.

I'd suggest to close this for now. We can always revisit if this proves to be a perf bottle neck in some scenarios or really requested API, for example.

**Comment by ManickaP** [comment]
After an internal discussion, I'll close this for now. We can always revisit this.
